### ISSUE TITLE
Status of v7

### ISSUE THREAD

**Issue by mschipperheyn** [sequelize/sequelize#15316]
## Issue Creation Checklist

[x ] I have read the [contribution guidelines](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)

## Issue Description

Is there any way to get some insight in the status of v7, particularly Typescript? I can see from all the ongoing work that A LOT of work has been done, which is awesome. My team is actually eagerly awaiting the chance to start migrating from sequelize-typescript because that library is blocking a number of upgrades. This will be quite a bit of work and I would like to get some idea of the release schedule, or key open issues. Is the api pretty much stable for instance? When can we expect something like a beta or release candidate? 

I was following some of the github project stuff before, but that was removed. 

Doing the migration, we might also be able to offer some feedback against a larger live code base. 

### Additional context

If the schedule for release aligns with our product release, we might actually migrate now instead of doing that on our own v2.

### Is this issue dialect-specific?

- [x ] No. This issue is relevant to Sequelize as a whole.
- [ ] Yes. This issue only applies to the following dialect(s): XXX, YYY, ZZZ
- [ ] I don't know.

### Would you be willing to resolve this issue by submitting a Pull Request?

- [ x] Yes, I have the time and I know how to start.
- [ ] Yes, I have the time but I don't know how to start, I would need guidance.
- [ ] No, I don't have the time, although I believe I could do it if I had the time...
- [ ] No, I don't have the time and I wouldn't even know how to start.

**Comment by WikiRik** [comment]
@ephys can say more about this, but the closest thing to a roadmap/key issues we've written down is https://github.com/orgs/sequelize/projects/1


The release plan is something like the following;
 Release a new alpha after #14259 where we release the rewrite on DataTypes as well. I think this should be a quite stable release.

Once we're happy with the features we have updated we'll do a v7 beta so the first projects can start upgrading and we can focus our efforts on bug fixing/documentation/other first-party packages (like CLI)

And then when we know that some projects have successfully upgraded and third-party packages have added v7 support (like sequelize-nest) we can release a full release (and deprecate v6 support a few months later)

**Comment by mschipperheyn** [comment]
Ok, I'd be up for doing a migration against a BETA

**Comment by mschipperheyn** [comment]
I mean beta (didn't mean that as a shout. Github edit seems to be failing)

**Comment by WikiRik** [comment]
We discussed this in our monthly meeting yesterday and we identified two key issues that we want to tackle before the v7 beta. Those are migrating the features of sequelize-typescript to sequelize directly, so decorators are supported natively and the usage of sequelize-typescript is no longer needed, and #14259
I'll close this issue in favour of https://github.com/sequelize/meetings/issues/23 and update that one accordingly soon.

We'll be in touch when the beta comes out to see if we can provide some support for your migration
